src/calculateRentalCost.js

```diff
@@ -4,7 +4,26 @@
  * @return {number}
  */
 function calculateRentalCost(days) {
-  // write code here
-}
+  let result = 0;
+  const DAY_COST = 40;
+  let discount = 0;
+
+  for (let n = 1; n <= days; n++) {
+    if (n >= 7) {
+      discount = 50;
+    }
+
+    if (n < 7) {
+      discount = 20;
+    }
 
+    if (n < 3) {
+      discount = 0;
+    }
+
+    result = n * DAY_COST - discount;
+  }
+
+  return result;
```

Comment:
The return statement correctly returns the calculated total cost. However, the logic leading up to this return needs to be corrected as per the previous comments.